### DISCUSSION TITLE
Supplementing freezed output with custom generator

### DISCUSSION THREAD

**Discussion by ryanhanks-bestow** [rrousselGit/freezed#722] [Q&A] [answered]
I'd like to be able to use the freezed data models (https://github.com/rrousselGit/freezed/blob/master/packages/freezed/lib/src/models.dart) to generate code to supplement my freezed classes.

Is this currently possible without forking freezed? Ideally I would be able to extend a custom Freezed builder. Would be nice if the output could be in the ".freezed" part, but it would be fine if it ended up elsewhere.

**Comment by rrousselGit** [comment, marked as answer]
I need to know more about your use-case.
Otherwise this is too broad to answer

**Reply by ryanhanks** [reply]
I want a generator to create an abstract handler class to handle instances of a union type.

```dart
@freezed
class Message with _$Message {
  factory Message.messageA() = MessageMessageA;

  factory Message.messageB() = MessageMessageB;
}

// Custom generator output

abstract class MessageHandlerBase<R> {
  R onMessageA(MessageMessageA value);

  R onMessageB(MessageMessageB value);

  R onMessage(Message message) =>
      message.map(messageA: onMessageA, messageB: onMessageB);
}

// My code that extends custom generator output

class MessageHandler extends MessageHandlerBase<int> {

  @override
  int onMessageA(MessageMessageA value) => 1;
    
  @override
  int onMessageB(MessageMessageB value) => 2;
}
```

I'm new to source_gen. I do see how this would be possible without using the freezed models, maybe I've overthought this..

**Reply by rrousselGit** [reply]
You shouldn't need the implementation details of Freezed for this.  
You should be able to obtain classes with the `@freezed` annotation and do whatever you want with them.

I'm not comfortable exposing the internals of Freezed. They aren't meant for public consumption
